### PR TITLE
[server] migrate ws without usageattribution

### LINKED ISSUE #17482: Very old pinned workspaces not shown after migration
Some old pinned workspaces have not been updated with an organizationId and are therefore not shown in the UI and not startable.

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

In the migration we didn't anticipate that we have only updated usageAttribuitionIds for workspace instances from last June until now.

This change makes sure that workspaces get an orgId oin migration and because we already have executed many migrations we unfortunately also need to fix this on the fly.

## Related Issue(s)

Fixes #17482
Fixes WEB-294
#### Preview status

gitpod:summary

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/gitpod-db/src/user-to-team-migration-service.spec.db.ts [hunk 1]
```diff
@@ -158,4 +158,41 @@ describe("Migration Service", () => {
         const teams = await teamDB.findTeamsByUser(user.id);
         expect(teams[0].name).to.be.eq("X Organization");
     });
+
+    it("should update 'organizationId' for workspace without attributionId", async () => {
```
Comment (svenefftinge) [inline components/gitpod-db/src/user-to-team-migration-service.spec.db.ts:162]: fails without the addition in the migration code

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 2]
```diff
@@ -1292,11 +1303,24 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
...
     protected async internalGetWorkspace(id: string, db: WorkspaceDB): Promise<Workspace> {
-        const ws = await db.findById(id);
-        if (!ws) {
+        const workspace = await db.findById(id);
+        if (!workspace) {
             throw new ResponseError(ErrorCodes.NOT_FOUND, "Workspace not found.");
         }
-        return ws;
+        if (!workspace.organizationId && this.user?.additionalData?.isMigratedToTeamOnlyAttribution) {
+            try {
```
Comment (svenefftinge) [inline components/server/src/workspace/gitpod-server-impl.ts:1311]: same here. Also only runs for workspaces without orgId and after migration

**Review by easyCZ — APPROVED** [review]
LGTM, thanks for the test

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 3]
```diff
@@ -1292,11 +1303,24 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
...
+                log.info({ userId: this.user.id }, "Updating workspace without orgId.");
+                const userOrg = await this.userToTeamMigrationService.getUserOrganization(this.user);
+                const latestInstance = await this.workspaceDb.trace({}).findCurrentInstance(workspace.id);
+                this.userToTeamMigrationService.updateWorkspacesOrganizationId(
```
Comment (AlexTugarev) [inline components/server/src/workspace/gitpod-server-impl.ts:1315]: Missing `await`? Or is this supposed to be not awaited on? 
https://github.com/gitpod-io/gitpod/pull/17489 to fix the build

**Pull request merged**